### PR TITLE
Avoid race condition in install.js when creating .git/hooks

### LINKED ISSUE #76: fs.existsSync followed by fs.mkdirSync fail with EEXIST
There is [a problem][1] with angular continuous integration. My suspicion is that the repository is created in between the calls to `fs.existsSync` and `fs.mkdirSync`, leading to failure. So, I am fixing our `install.js` script to "ask forgiveness not permission" like so:

```
function xmkdirSync (path, mode) {
  try {
    fs.mkdirSync(path, mode)
  } catch (err) {
    if (/EPERM/.test(err.message)) {
      console.error('Failed to create: ' + git)
      console.error('Make sure you have the necessary permissions')
    } else if (/EEXIST/.test(err.message)) {
      // do nothing
    } else {
      throw err
    }
  }
}
```

Is that ok? Is there perhaps some other issue I am missing?

[1]: https://circleci.com/gh/angular/angular/3788

### PR DESCRIPTION
**Pull request by all3fox**
Closes #76 

I am not sure how to properly exit from a `catch` block: with `process.exit(1)` or rethrowing the error.

### CODE REVIEW

File: install.js [hunk 1]
```diff
@@ -9,13 +9,27 @@ if (options && options.nohook) {
   process.exit(0)
 }
 
+function xmkdirSync (path, mode) {
+  try {
+    fs.mkdirSync(path, mode)
+  } catch (err) {
+    if (/EPERM/.test(err.message)) {
+      console.error('Failed to create: ' + git)
```
Comment (jzaefferer) [inline install.js]: Wrong var? Looks like it should be `path`

**Comment by jzaefferer** [comment]
I'm not sure either, but doing both in one method is questionable. Since we're using `process.exit(0)` already,  I suggest we stick with that.

**Comment by all3fox** [comment]
I have prepublished this as `2.7.1-0` under `beta` tag. This PR fixes Angular's CircleCI failure (it still fails but bacause of `gulp lint` on some test, not because of us). What do you think about merging it? Does it solve your CircleCI problem?

**Comment by jzaefferer** [comment]
Just tested this, works well! Even as a pre-push hook - this is what I added to the readme for the project: https://gist.github.com/jzaefferer/a4b588abb91e6a020c28aa07880f16a3 - might be useful for other projects as well.

**Comment by all3fox** [comment]
Glad to hear it! Will add that bit to the README, then merge this branch, create `2.7.1`, publish on npm and then go to update angular PRs.

**Pull request merged**